File: LeetCode75_GreatestCommonDivisorOfStrings.java
class LeetCode75_GreatestCommonDivisorOfStrings {
    public static void main(String[] args)
    {
        String word1 = "ABABAB";
        String word2 = "ABAB";
        String find_result = gcdOfStrings(word1,word2);
        System.out.println(find_result);
    }

    public static String gcdOfStrings(String str1, String str2) {
        String result = "";
        if((str1+str2).equals(str2+str1))
        {
            int common = findGCD(str1.length(),str2.length());
            for(int i = 0; i< common; i++)
            {
                result += str1.charAt(i);
            }
        }
        return result;
    }

    public static int findGCD(int a, int b){
        while(b!=0)
        {
            int temp = b;
            b = a%b;
            a = temp;
        }
        return a;
    }

}

File: LeetCode75_GreatestCommonDivisorOfStrings.py
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
       if(str1 + str2 != str2 + str1):
           return ""
       else:
           from math import gcd
           return str1[:gcd(len(str1), len(str2))]

    


if __name__ == "__main__":
    solution = Solution()
    str1 = "ABCABCABCABC"
    str2 = "ABCABC"
    print(solution.gcdOfStrings(str1, str2))
